### ISSUE TITLE
bug: Potentially unexpected output given the "*" operator

### ISSUE THREAD

**Issue by dpchamps** [messageformat/messageformat#361]
While authoring https://github.com/messageformat/messageformat/pull/360, I noticed a potentially unexpected edge case while using the `*` operator:

Let's say I'm authoring some messages, and I want to be overly terse:

```
const messagePacks = {
    "en": {
      es: "Estimation Date: {date, date, ::EEEMMMd}",
      fi: "Financial Incentive Date: {date, date, ::EEEMMMd}",
    },
  };
```

If I were to compile this against the `new MessageFormat("*")` invocation, I'm going to get _very_ unexpected output. Namely, spanish and finnish localized dates.

See the repro here: https://github.com/dpchamps/mf-locale-collision-repro

Idea for a fix: we could add an optional arg to `compile` to indicate `isRecursing`, and then not pull from the plurals at all.

**Comment by eemeli** [comment]
Yeah, this feels like a bug, and rather unlikely to affect real-world users in a way that anyone would (should?) depend on.
